internal/expr/bindinputs.go

```diff
@@ -29,12 +29,24 @@ func (tbe *TypeBoundExpr) BindInputs(args ...any) (pq *PrimedQuery, err error) {
 	for _, arg := range args {
 		v := reflect.ValueOf(arg)
 		if v.Kind() == reflect.Invalid || (v.Kind() == reflect.Pointer && v.IsNil()) {
-			return nil, fmt.Errorf("need struct or map, got nil")
+			return nil, fmt.Errorf("need valid value, got nil")
 		}
 		v = reflect.Indirect(v)
 		t := v.Type()
-		if v.Kind() != reflect.Struct && v.Kind() != reflect.Map {
-			return nil, fmt.Errorf("need struct or map, got %s", t.Kind())
+		switch v.Kind() {
+		case reflect.Map, reflect.Slice:
+			if v.IsNil() {
+				return nil, fmt.Errorf("need valid %s, got nil", t.Kind())
+			}
+			if t.Name() == "" {
+				return nil, fmt.Errorf("cannot use anonymous %s", t.Kind())
+			}
+		case reflect.Struct:
+			if t.Name() == "" {
+				return nil, fmt.Errorf("cannot use anonymous %s", t.Kind())
+			}
+		default:
+			return nil, fmt.Errorf("need supported value, got %s", t.Kind())
```

Comment:
Yea, that sounds good. I'd also like to replace this in the future with a list of the types from `bindtypes` since at this stage we do know exactly what we're looking for.